### PR TITLE
Vectorfield refactors

### PR DESCRIPTION
**Pull request by MukundhMurthy**
updated docstring to correct format and ensured typing/docstring are added to all user relevant functions

### CODE REVIEW

File: dynamo/vectorfield/vector_calculus.py [hunk 1]
```diff
@@ -52,54 +54,56 @@
...
-            vf_dict["method"] = "sparsevfc"
-        if vf_dict["method"].lower() == "sparsevfc":
-            vector_field_class = SvcVectorField()
-            vector_field_class.from_adata(adata, basis=basis)
-        elif vf_dict["method"].lower() == "dynode":
-            vf_dict["parameters"]["load_model_from_buffer"] = True
-            vector_field_class = vf_dict["dynode_object"]  # dynode_vectorfield(**vf_dict["parameters"])
-        else:
-            raise ValueError("current only support two methods, SparseVFC and dynode")
+        vector_field_class = vector_field_class = get_vf_class(adata, basis=basis)
```
Comment (Sichao25) [inline dynamo/vectorfield/vector_calculus.py:106]: Hi, @MukundhMurthy. Is this a typo? Also see line 166.

Comment (MukundhMurthy) [reply dynamo/vectorfield/vector_calculus.py]: Yes, that was a typo – I had written the `get_vf_class` function to get rid of many duplicated code segments

File: dynamo/vectorfield/vector_calculus.py [hunk 2]
```diff
@@ -1109,53 +925,34 @@ def curl(adata, basis="umap", vector_field_class=None, method="analytical", **kw
 
 
 def divergence(
-    adata,
-    cell_idx=None,
-    sampling=None,
-    sample_ncells=1000,
-    basis="pca",
+    adata: AnnData,
+    cell_idx: bool = None,
```
Comment (Sichao25) [inline dynamo/vectorfield/vector_calculus.py:929]: The description of this function misses a few Args, and the type of cell_idx/sampling is not bool. The same problem can be found on line 330. I recommend using` Optional[Literal[...]]` for parameters like sampling.

File: dynamo/vectorfield/scVectorField.py [hunk 3]
```diff
@@ -237,15 +237,15 @@ def get_P(Y, V, sigma2, gamma, a, div_cur_free_kernels=False):
...
-    k=30,
-    distance_free=True,
-    n_int_steps=20,
-    cores=1,
-):
+    k: int = 30,
+    distance_free: bool = True,
+    n_int_steps: int = 20,
+    cores: int = 1,
+) -> Tuple[np.ndarray, NearestNeighbors]:
```
Comment (Sichao25) [inline dynamo/vectorfield/scVectorField.py:248]: Could the output be an NNDescent type? Also see line 585.

**Comment by Sichao25** [comment]
Hi @MukundhMurthy, I am going to make some changes to the issues I mentioned above. Let me know if you don't like those changes.

**Comment by MukundhMurthy** [comment]
@Sichao25 Those changes all make sense, thank you for catching them!

**Comment by Sichao25** [comment]
This pull request looks good to me.

**Comment by Xiaojieqiu** [comment]
@Sichao25 please fix my above review comments

File: dynamo/vectorfield/Ao.py [hunk 4]
```diff
@@ -131,23 +110,3 @@ def Ao_pot_map(vecFunc, X, D=None, **kwargs):
...
-    m = int(ndim * (ndim - 1) / 2)
-    q0 = np.ones(m) * np.mean(np.diag(D)) * 1000
-    for i in tqdm(range(nobs), "Calc Ao potential"):
-        X_s = X[i, :]
-        F = fjac(X_s)
-        Q, _ = solveQ(D, F, q0=q0, **kwargs)
-        H = np.linalg.inv(D + Q).dot(F)
-        U[i] = -0.5 * X_s.dot(H).dot(X_s)
-
-    return U.flatten()
```
[the remark this answers is not in the document]
Comment (Sichao25) [reply dynamo/vectorfield/Ao.py:153]: Do we still need this function? It is not used by other functions.

**Comment by Sichao25** [comment]
@Xiaojieqiu They are done.

**Pull request merged**